pkg/cloudprovider/provider/azure/provider.go

```diff
@@ -476,7 +490,7 @@ func (p *provider) Create(machine *v1alpha1.Machine, data *cloudprovidertypes.Pr
 	tags[machineUIDTag] = to.StringPtr(string(machine.UID))
 
 	adminUserName := string(providerCfg.OperatingSystem)
```

Comment:
kinda funny Operating System variable name :)

Probably getting default / hardcoded login based on the input OS deserves dedicated public function

Reply:
I am not sure tbh since my PR has nothing to do with it :). However, why do you think this needs a public function?

Reply:
even if public function is questionable, the var name `adminUserName` which actually hold name of the OS but not username, needs to be changed (but adminUserName is also needed of course).

Reply:
The variable does not hold the OS name, it holds the user name. The user name is just initialized with the OS name. It makes perfect sense. By your logic you cannot have `a := 3`, but it needs to be `three := 3` ;-)

Reply:
Yeah that's correct. I didn't like it either tbh but I didn't want to fiddle with something that my task has nothing to do with and it is not failing. However, since someone shed the light on it I will change it into a private function

Reply:
@xrstf what @kron4eg meant is, having a dedicated function for this would be cleaner. because if you take a look of how this adminUserName is used and validated can be moved into a function called e.g: osUsername(), and there the validation is done too